### PR TITLE
bugfix/WAR-1514: added documentation to the code done in war-1156

### PR DESCRIPTION
**Pull request by navvrnd**
Added documentation for the code done in war-1156.

### CODE REVIEW

**Comment by navvrnd** [comment]
Hi @kaheichan Why can't this be pushed for this release? Please let me know, if there is any thing pending from my side.

File: warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py [hunk 1]
```diff
@@ -312,11 +312,15 @@ def print_warn_msg(keyvars, numpats):
...
+                # get the patterns from pattern entries in testdata file
                 patterns = [k.get("resp_pattern_req") for k in resp_keys]
+                # warn if number of patterns does not match number of resp_ref keys
                 if len(keys) != len(patterns):
                     print_warn_msg(keys, len(patterns))
                 if inorder:
                     pNote(save_msg1+' inorder.')
+                    # since inorder pattern matching selected, join all the
+                    # patterns in order to create a single big pattern
                     cpatterns = map(lambda s: "(" + s + ")", patterns)
```
Comment (kaheichan) [inline warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py]: lambda function can be replaced by list comprehension, which is more readable

File: warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py [hunk 2]
```diff
@@ -325,8 +329,11 @@ def print_warn_msg(keyvars, numpats):
                     reobj = re.search(pattern, response, re.DOTALL)
                     if reobj:
                         grps = reobj.groups()
+                        # update response_dict with resp_ref keys and
+                        # their corresponding matched patterns
                         response_dict.update(dict(zip(keys, grps)))
                         pNote(save_msg2.format(pattern))
+                        # print to console the key and the corresponding match stored
                         map(lambda x: pNote(save_msg3.format(*x)), zip(keys,
```
Comment (kaheichan) [inline warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py]: same here, please use comprehension

File: warrior/Framework/Utils/data_Utils.py [hunk 3]
```diff
@@ -468,18 +468,22 @@ def _get_cmd_details(testdata, global_obj, system_name,
...
+                the testdata or global section. Return None if it is not
+                found in both testdata and global section
                 """
                 if testdata.find(key) is not None:
                     return testdata.find(key)
                 if global_obj is not None:
                     global_key_elem = global_obj.find("keys")
                     global_keys = xml_Utils.get_child_node_list(global_key_elem)
+                    # return the first matched entry in global section with tag=key
                     return filter(lambda gk: gk.tag == key, global_keys)[0]
```
Comment (kaheichan) [inline warrior/Framework/Utils/data_Utils.py]: also in here

File: warrior/Framework/Utils/data_Utils.py [hunk 4]
```diff
@@ -468,18 +468,22 @@ def _get_cmd_details(testdata, global_obj, system_name,
...
                 return None
             keylist = _get_cmdparams_list(testdata, global_obj, attrib)[0]
             if keylist:
+                # get the keys to be used for pattern matching in this command
                 keys = map(lambda x: x.strip(), keylist.split(','))
```
Comment (kaheichan) [inline warrior/Framework/Utils/data_Utils.py]: and here

File: warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py (hunk 1 above)
Comment (navvrnd) [reply warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py]: done.

File: warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py (hunk 2 above)
Comment (navvrnd) [reply warrior/Framework/ClassUtils/WNetwork/warrior_cli_class.py]: done.

File: warrior/Framework/Utils/data_Utils.py (hunk 3 above)
Comment (navvrnd) [reply warrior/Framework/Utils/data_Utils.py]: done

File: warrior/Framework/Utils/data_Utils.py (hunk 4 above)
Comment (navvrnd) [reply warrior/Framework/Utils/data_Utils.py]: done

**Comment by navvrnd** [comment]
@kaheichan made the requested changes and added full regression log in the ticket. Please check and merge.

**Comment by kaheichan** [comment]
Saw the regression log and the code change looks good

**Pull request merged**